src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,518 @@
+            else
+            {
+                logger.warn("Dropping {} Polygon {}", polygonType, polygon);
+                return false;
+            }
+        };
+    }
+
+    private AtlasEntityPolygonsFilter(final Type filterType, final Collection<Polygon> polygons,
+            final Collection<MultiPolygon> multiPolygons)
+    {
+        this(filterType, DEFAULT_INTERSECTION_POLICY, polygons, multiPolygons);
+    }
+
+    private AtlasEntityPolygonsFilter(final Type filterType,
+            final IntersectionPolicy intersectionPolicy, final Collection<Polygon> polygons,
+            final Collection<MultiPolygon> multiPolygons)
+    {
+        this.filterType = filterType;
+        this.intersectionPolicy = intersectionPolicy;
+        final QuadTree<Located> vettedPolygons = new QuadTree<>();
+        if (filterType == Type.INCLUDE)
+        {
+            this.includePolygons.addAll(polygons == null ? Collections.EMPTY_SET
+                    : polygons.stream().filter(notOverlappingPolygon(vettedPolygons, Type.INCLUDE))
+                            .collect(Collectors.toSet()));
+            this.includeMultiPolygons.addAll(multiPolygons == null ? Collections.EMPTY_SET
+                    : multiPolygons.stream()
+                            .filter(notOverlappingMultipolygon(vettedPolygons, Type.INCLUDE))
+                            .collect(Collectors.toSet()));
+        }
+        else
+        {
+            this.excludePolygons.addAll(polygons == null ? Collections.EMPTY_SET
+                    : polygons.stream().filter(notOverlappingPolygon(vettedPolygons, Type.EXCLUDE))
+                            .collect(Collectors.toSet()));
+            this.excludeMultiPolygons.addAll(multiPolygons == null ? Collections.EMPTY_SET
+                    : multiPolygons.stream()
+                            .filter(notOverlappingMultipolygon(vettedPolygons, Type.EXCLUDE))
+                            .collect(Collectors.toSet()));
+        }
+    }
+
+    @Override
+    public boolean test(final AtlasEntity object)
+    {
+        return noPolygons().or(isIncluded()).or(isNotExcluded()).test(object);
+    }
+
+    private boolean excludePolygonsEmpty()
+    {
+        return this.excludePolygons.isEmpty() && this.excludeMultiPolygons.isEmpty();
+    }
+
+    private boolean includePolygonsEmpty()
+    {
+        return this.includePolygons.isEmpty() && this.includeMultiPolygons.isEmpty();
+    }
+
+    private Predicate<AtlasEntity> isIncluded()
+    {
+        return entity -> this.filterType == Type.INCLUDE && (this.includePolygons.stream().anyMatch(
+                polygon -> this.intersectionPolicy.polygonEntityIntersecting(polygon, entity))
+                || this.includeMultiPolygons.stream()
+                        .anyMatch(multiPolygon -> this.intersectionPolicy
+                                .multiPolygonEntityIntersecting(multiPolygon, entity)));
+    }
+
+    private Predicate<AtlasEntity> isNotExcluded()
```

Comment:
Because the filter is only ever one type, there is no consideration of inclusion in this function. This function only returns true if it's an exclude filter and the Entity is not excluded.